src/programs/example.tsx

```diff
@@ -0,0 +1,73 @@
+const jsonVal = require('./data.json');
+const _ = require('lodash');
+
+
+interface ConditionProps {
+    [index: number]: any | any[];
+}
+
+const isOperator = (val: string) => ['not', 'or', 'and'].includes(val);
+const filterOperator = (conditions: ConditionProps) => {
+
+    return _.find(conditions, (condition: ConditionProps) => {
+        if ((typeof (condition) === 'string') && (isOperator(condition))) {
+            return condition;
+        };
+    });
+};
+
+const filterCriterias = (conditions: ConditionProps) => {
+    const criterias: ConditionProps[] = [];
+    _.forEach(conditions, (condition: ConditionProps[]) => {
+        if (Array.isArray(condition) || !isOperator(condition)) {
+            criterias.push(condition);
+        }
+    })
+    return criterias;
+}
+
+const hasCondition = (item: any, criteria: ConditionProps) => {
+    const hasOperator = filterOperator(criteria);
+    if (hasOperator) {
+        const nextCri = filterCriterias(criteria);
+        return applyCondition(item, hasOperator, nextCri)
+    }
+    const key = criteria[0];
+    const values = criteria[1];
+    const nKey = key === 'from_country' ? 'country' : 'gender_is' ? 'gender' : key;
```

Comment:
There should a common function that returns the desired output. When reading a complex data adding conditions wouldn't be scale-able. Writing functions will help us to test the code better when in development.